File: vente-plus/src/app/models/produit.ts
import { Categorie } from "./categorie";

export interface Produit {
    id?: number;
    nom: string;
    prix: number;
    quantite: number;
    reference: string;
    imageUrl?: string;
    createdAt?: Date;
    categorieId?: number;
    categorie?: Categorie;
}

File: vente_plus_back/dashbord-service/src/main/java/org/ms/dashbordservice/model/Produit.java
package org.ms.dashbordservice.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Produit {
    private Long id;
    private String nom;
    private BigDecimal prix;
    private long quantite;
    private String reference;
    private String imageUrl;
    private LocalDateTime createdAt;
    private Long categorieId;
    private Categorie categorie;
}
